src/flow/darcy_flow_assembly.hh

```diff
@@ -150,7 +150,9 @@ public:
         assemble_element(ele_ac);
         assemble_source_term(ele_ac);
         
-        
+        //due to time dependent flow, fill the whole diagonal (otherwise it is allocated)
```

Comment:
So it is necessary to keep matrix structure unchanged? But what can change for unsteady case? Only possibility is switching from steady to unsteady. Comment seems misleading since this have noting to do with allocation, but with allowing reuse of matrix structure.

Reply:
This was a fast fix:
- in steady case, we want the zeros on the diagonal to be ignored (and not allocated)
- in unsteady case, if the zeros on the diagonal are ignored (and not allocated), then it crashes when assembling time terms (in `setup_time_term()`) because we need to write to the diagonal and not enough memory is allocated in that rows..

Any better suggestions, what to do with it?